src/common/dsp/SurgeVoice.cpp

```diff
@@ -992,13 +992,63 @@ bool SurgeVoice::process_block(QuadFilterChainState &Q, int Qe)
     {
         if (is_wide)
         {
-            mul_block(osc[0]->output, osc[1]->output, tblock, BLOCK_SIZE_OS_QUAD);
-            mul_block(osc[0]->outputR, osc[1]->outputR, tblockR, BLOCK_SIZE_OS_QUAD);
+            switch (scene->level_ring_12.deform_type) {
```

Comment:
So my primary thought here is: Do we want to add a function like `SurgeVoice::ringModSources(o1L, o1R, o2L, o2R, rR, rL, is_wide, type)` so we don't have to copy this block twice?

Reply:
yeah probably worth doing

Reply:
should that function be in basic_dsp.h along with what it calls?

Reply:
no i would make it a new member on SurgeVoice which calls the basic_dsp methods myself.

Reply:
basic_dsp is really block operations. The switch and is wide check belongs in the voice is my thinking.

Reply:
working on this now